src/pubkey.cpp

```diff
@@ -181,6 +182,22 @@ int ecdsa_signature_parse_der_lax(secp256k1_ecdsa_signature* sig, const unsigned
     return 1;
 }
 
+/** Nothing Up My Sleeve (NUMS) point
+ *
+ *  NUMS_H is a point with an unknown discrete logarithm, constructed by taking the sha256 of 'g'
+ *  after DER encoding (without compression), which happens to be a point on the curve.
+ *
+ *  More precisely, H is derived by running the following script with the sage mathematics software:
+
+    import hashlib
+    F = FiniteField (0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFEFFFFFC2F)
+    G_DER = '0479be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798483ada7726a3c4655da4fbfc0e1108a8fd17b448a68554199c47d08ffb10d4b8'
+    G2 = EllipticCurve ([F (0), F (7)]).lift_x(F(int(hashlib.sha256(bytes.fromhex(G_DER)).hexdigest(),16)))
+    print('%x %x' % G2.xy())
```

Comment:
would it make sense to add a Python test which runs this exact script and checks that the result equals `NUMS_H_DATA`?
Maybe there's something like this already, found parts of it, could you please check? In that case we wouldn't need to put code into a comment, we could just point to the test instead.

Reply:
I think that would require adding [sage](https://www.sagemath.org) to our test dependencies.

Reply:
I don't see the value of running a test like this? The code here is not the definition of `H`, just one way of calculating the value, based on the definition in BIP341. For example, if the test were to fail, that would just mean the comment is out of date, not that our value for `H` is incorrect.

Reply:
What if we just extract it to code like it was done with the [rest of the sage files](https://github.com/search?q=repo%3Abitcoin%2Fbitcoin%20.sage&type=code), and reference the file instead of the code (I don't like dead code in comments), e.g. https://github.com/bitcoin/bitcoin/blob/master/src/secp256k1/src/ecdsa_impl.h#L19 ?

Reply:
Going to leave this as is for now. The sage code in the comment is meant to provide background on where the constant comes from, so I'd prefer to leave it since it is more precise than a written out explanation. However, the code is not essential for checking the "correctness" of the value in that it would be sufficient for someone to simply check that this value matches BIP341. This value will never change, either, so I don't see any advantage to having a sage file in this repo for generating the constant.

Reply:
Thanks for the code snippets @sipa . Seems like there is appetite for this, so added this script to `test/functional/test_framework/crypto/secp256k1.py` as a unit test and removed the sage code from the comment.